festim/exports/txt_export.py

```diff
@@ -72,62 +117,40 @@ def write(self, current_time, steady):
             if not os.path.exists(dirname):
                 os.makedirs(dirname, exist_ok=True)
 
-            # if steady or it is the first time to export
-            # write data
+            # create header if steady or it is the first time to export
             # else append new column to the existing file
-            if steady or self._first_time:
-                if steady:
-                    header = "x,t=steady"
+            if final_time is None or self._first_time:
+                if final_time is None:
```

Comment:
Just for readability, it would be clearer to keep the `steady` variable here but add

```suggestion
            steady = final_time is None
            if steady or self._first_time:
                if steady:
```